modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/translator/DeepLTranslator.java

```diff
@@ -0,0 +1,249 @@
+
+		for (Map.Entry<String, String> entry : fieldsMap.entrySet()) {
+			translatedFieldsMap.put(
+				entry.getKey(),
+				_translate(
+					deepLTranslatorConfiguration.url(),
+					deepLTranslatorConfiguration.authKey(), entry.getValue(),
+					sourceLanguageCode, targetLanguageCode));
+		}
+
+		return new TranslatorPacket() {
+
+			@Override
+			public long getCompanyId() {
+				return translatorPacket.getCompanyId();
+			}
+
+			@Override
+			public Map<String, String> getFieldsMap() {
+				return translatedFieldsMap;
+			}
+
+			@Override
+			public String getSourceLanguageId() {
+				return translatorPacket.getSourceLanguageId();
+			}
+
+			@Override
+			public String getTargetLanguageId() {
+				return translatorPacket.getTargetLanguageId();
+			}
+
+		};
+	}
+
+	@Activate
+	@Modified
+	protected void activate(Map<String, Object> properties) {
+		_deepLTranslatorConfiguration = ConfigurableUtil.createConfigurable(
+			DeepLTranslatorConfiguration.class, properties);
+	}
+
+	private String _getLanguageCode(String languageId) {
+		List<String> list = Arrays.asList(
+			StringUtil.split(languageId, CharPool.UNDERLINE));
+
+		return StringUtil.toUpperCase(list.get(0));
+	}
+
+	private List<String> _getSupportedLanguages(
+		DeepLTranslatorConfiguration deepLTranslatorConfiguration) {
+
+		try {
+			List<SupportedLanguage> supportedLanguages =
+				_deepLClient.getSupportedLanguages(
+					deepLTranslatorConfiguration.authKey(),
+					DeepLConstants.TARGET,
+					DeepLConstants.SUPPORTED_LANGUAGES_URL);
+			List<String> languages = new ArrayList<>();
+
+			supportedLanguages.forEach(
+				supportedLanguage -> languages.add(
+					supportedLanguage.getLanguage()));
+
+			return languages;
+		}
+		catch (IOException ioException) {
+			_log.error(
+				"Failed to call supported language list." +
+					System.lineSeparator() + ioException.getLocalizedMessage());
+
+			return new ArrayList<>();
+		}
+		catch (TranslatorException translatorException) {
+			throw new RuntimeException(translatorException);
+		}
+		catch (PortalException portalException) {
+			throw new RuntimeException(portalException);
+		}
+	}
+
+	private String _translate(
+			String url, String authKey, String text, String sourceLanguageCode,
+			String targetLanguageCode)
+		throws TranslatorException {
+
+		try {
+			if (Validator.isBlank(text)) {
+				return text;
+			}
+
+			TranslateResponse translateResponse = _deepLClient.execute(
+				authKey, text, sourceLanguageCode, targetLanguageCode, url);
+
+			List<Translation> translations =
+				translateResponse.getTranslations();
+
+			Translation translation = translations.get(0);
+
+			return translation.getText();
+		}
+		catch (IOException ioException) {
+			throw new TranslatorException(
+				"DeepL translator returns original text. " +
+					ioException.getLocalizedMessage());
+		}
+		catch (PortalException portalException) {
+			throw new RuntimeException(portalException);
+		}
```

Comment:
The caller already declares `throws PortalException`, so there's no need to wrap it in a runtime exn.